src/node/rpc/handlerregistry.h

```diff
@@ -68,80 +129,16 @@ namespace ccf
      *
      * @param method Method name
      * @param f Method implementation
-     * @param rw Flag if method will Read, Write, MayWrite
-     * @param params_schema JSON schema for params object in requests
-     * @param result_schema JSON schema for result object in responses
-     * @param require_client_signature If true, client request must be signed
-     * @param execute_locally If true, request is executed without consensus
-     * (PBFT only)
+     * @param read_write Flag if method will Read, Write, MayWrite
      */
-    void install(
-      const std::string& method,
-      HandleFunction f,
-      ReadWrite rw,
-      const nlohmann::json& params_schema = nlohmann::json::object(),
-      const nlohmann::json& result_schema = nlohmann::json::object(),
-      bool require_client_signature = false,
-      bool execute_locally = false)
-    {
-      handlers[method] = {f,
-                          rw,
-                          params_schema,
-                          result_schema,
-                          require_client_signature,
-                          execute_locally};
-    }
-
-    void install(
-      const std::string& method,
-      HandleFunction f,
-      ReadWrite rw,
-      bool require_client_signature)
-    {
-      install(
-        method,
-        f,
-        rw,
-        nlohmann::json::object(),
-        nlohmann::json::object(),
-        require_client_signature);
-    }
-
-    template <typename In, typename Out, typename F>
-    void install_with_auto_schema(
-      const std::string& method,
-      F&& f,
-      ReadWrite rw,
-      bool require_client_signature = false,
-      bool execute_locally = false)
-    {
-      auto params_schema = nlohmann::json::object();
-      if constexpr (!std::is_same_v<In, void>)
-      {
-        params_schema = ds::json::build_schema<In>(method + "/params");
-      }
-
-      auto result_schema = nlohmann::json::object();
-      if constexpr (!std::is_same_v<Out, void>)
-      {
-        result_schema = ds::json::build_schema<Out>(method + "/result");
-      }
-
-      install(
-        method,
-        std::forward<F>(f),
-        rw,
-        params_schema,
-        result_schema,
-        require_client_signature,
-        execute_locally);
-    }
-
-    template <typename T, typename... Ts>
-    void install_with_auto_schema(const std::string& method, Ts&&... ts)
+    Handler& install(
```

Comment:
Might be worth documenting the new `@return` here?

Reply:
Added.